### DISCUSSION TITLE
Drafted release notes should be public instead of private

### DISCUSSION THREAD

**Discussion by ssbarnea** [community/community#4192] [Other Feature Feedback, Questions, & Ideas] [answered]
I do really think that draft releases should be public as there is no real reason for preventing people from outside the project team to see them before the release happens.

I help maintaining a (too) wide range of open-source projects and on some of them I do not even have core access. How can a user know which fixes would go into the next release if they do not see any draft when they visit the releases page? They have no clue about what was already merged and is expected to be part of the next release.

One very useful action used in conjunction with releases is the [release drafter](https://github.com/marketplace/actions/release-drafter), which builds awesome release notes. Still, due to the fact that drafts are private that information is not available to the users.

**Comment by MylesBorins** [comment]
Hey @ssbarnea thanks for the suggestion.

While I agree that this is not perfect I can think of some reasons to keep a draft private such as wanting to review copy before publishing or drafting a security release.

For other projects I've taken advantage of `pre-releases` and `release-branches` to allow folks to know what's coming down the pipeline. `pre-releases` in particular are a great way to give folks a release-candidate or alpha that they can take for a spin.

At this time I don't think we'll be looking to change the default of drafts to be public, and thinking through the UX of allow folks to decide if drafts are public or not, and where they should be visible feels a bit overly complicated.

Do you think that you would be able to utilize `pre-releases` to accomplish giving folks an earlier idea of what might be in a release?

**Reply by ssbarnea** [reply]
pre-release would not help here, they are very close to releases from this point of view. You need to release them before someone can see them. Even when using them you have the same issue; how to make the list of changes merged since the pre-release visible to the visitor? That is a place were in-repo changlog rules, is is as visible as possible... only that it requires very good review/merge habits in order to be kept in sync. That is not perfect either as it causes merge conflicts between PRs.

Keep in mind that vast majority of opensource projects are in a poor state of maintenance, where the few people that can produce a release can barely find time to review PRs or spend time planning a release. Most of these to not have any schedule plan because is impossible to have one. Usually a release happens when an urgent issue appears, more like a pressure release valve. I know projects that failed to make a release in 2 years, cookiecutter is such an example.

As many changes pile up, we need to make those visible to the users.

In fact lack of visibility for draft/pending releases is one major reason for people to avoid using github releases and stick to other approaches like maintaining an in repo changelog.md or building one using fragments (towncrier or reno).

What makes the experience even more awkward is that you cannot ask other contributors to review a draft because you may not know if they can even see the draft. On many opensource project you have no clue who has enough access to read or edit the draft. I often contribute to projects where I have no clue who is core (active core even more tricky).

**Reply by MylesBorins** [reply]
@ssbarnea I'll continue to think on this but at this time it is going to be very difficult to change the defaults of draft releases to be visible without breaking the various valid use cases for keeping the draft private.

We are actively working on features that will make the process of creating releases + release notes substantially easier for developers and hopefully this can help with some of the problems you've outlined.

As there currently is no way to make commits or changes as part of the release in GitHub releases I think there are some challenges with treating it as a source of truth for "in process" releases. This is something we are also taking a look at, but don't currently have solutions.

One solution that I particularly like for "in-process" releases is [release please](https://github.com/googleapis/release-please) from the google cloud team. Have you seen it before?

**Reply by achimnol** [reply]
> We are actively working on features that will make the process of creating releases + release notes substantially easier for developers and hopefully this can help with some of the problems you've outlined.

I'd love to see these improvements soon!

**Comment by MylesBorins** [comment, marked as answer]
After more consideration we have decided to not move forward with changing the defaults on drafts. Currently drafts are private and many people expect that behavior, as such a draft release might include private details and potentially fixes for 0day security vulnerabilities. Changing this behavior could not only be disruptive, but dangerous.

We do already have pre-releases which are a way to create candidates that you can share with folks.

**Reply by HealthyPear** [reply]
but why not at least add an option tickbox to let the maintainers decide if they want to share it publicly? this helps users understand at a glance what has been changed since the latest release

**Reply by zyv** [reply]
I'd like to second that...a setting to make drafts public solves the problem and doesn't introduce any compatibility issues. It's a shame that we still don't have a way to make drafts public.

**Comment by guizmaii** [comment]
@MylesBorins Could this be a configuration? 
I understand and agree with your point about keeping the default as "private" but I don't understand why it's you who should decide if our Draft releases are public or private.

For some projects, I agree it makes sense to have draft release private but for some other projects, like the open-source projects I maintain and others I contribute to, it hides things from our users which have more difficulty understanding what has been merged and what will be released soon

We use release-drafter to automatically maintain our Draft releases up-to-date and it would be very useful and provide better visibility to the users of our open-source libraries if we could show these Draft release to everyone.

**Reply by gdams** [reply]
I'd like to second this, it would be very useful to be able to opt into public draft releases on a per repo configuration.

**Reply by MylesBorins** [reply]
I don't work at GitHub anymore, apologies I can't help with this.
